From c755ec847459d9d3807c22c2d4d6aa6f9c86dc7c Mon Sep 17 00:00:00 2001
From: John Budorick <jbudorick@chromium.org>
Date: Tue, 19 Nov 2019 22:24:20 +0000
Subject: [PATCH] 3904: chrome_cleaner: ensure internal targets are reachable
 from :gn_all.

(cherry picked from commit eee513dfc780ad4bcaa74b67f2e0eaebf57b5c93)

Bug: 984162, 1004848
Change-Id: Ibb7c47f0188de8e077cd30aa8d68ca61133d2037
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/1904549
Commit-Queue: John Budorick <jbudorick@chromium.org>
Reviewed-by: Joe Mason <joenotcharles@chromium.org>
Cr-Original-Commit-Position: refs/heads/master@{#713683}
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/1922299
Cr-Commit-Position: refs/branch-heads/3904@{#895}
Cr-Branched-From: 675968a8c657a3bd9c1c2c20c5d2935577bbc5e6-refs/heads/master@{#693954}
---
 BUILD.gn                       |  2 +-
 chrome/chrome_cleaner/BUILD.gn | 23 ++++++++++++++++++++---
 2 files changed, 21 insertions(+), 4 deletions(-)

diff --git a/BUILD.gn b/BUILD.gn
index 5e2f6b9b0f41c..ac0297608594f 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -604,7 +604,7 @@ group("gn_all") {
   if (is_win) {
     deps += [
       "//base:pe_image_test",
-      "//chrome/chrome_cleaner:chrome_cleaner_unittests",
+      "//chrome/chrome_cleaner",
       "//chrome/chrome_elf:chrome_elf_unittests",
       "//chrome/chrome_elf:dll_hash_main",
       "//chrome/elevation_service:elevation_service_unittests",
diff --git a/chrome/chrome_cleaner/BUILD.gn b/chrome/chrome_cleaner/BUILD.gn
index fbd2d8d5307aa..1671a3f7d097c 100644
--- a/chrome/chrome_cleaner/BUILD.gn
+++ b/chrome/chrome_cleaner/BUILD.gn
@@ -64,9 +64,6 @@ static_library("other_executable_definitions") {
 }
 
 test("chrome_cleaner_unittests") {
-  # Make this target findable from the "all" target used by the builders.
-  visibility += [ "//.:gn_all" ]
-
   sources = [
     "//chrome/chrome_cleaner/test/test_main.cc",
   ]
@@ -126,3 +123,23 @@ test("chrome_cleaner_unittests") {
     ]
   }
 }
+
+group("chrome_cleaner") {
+  testonly = true
+
+  # Make this target findable from the "all" target used by the builders.
+  visibility += [ "//.:gn_all" ]
+
+  deps = [
+    ":chrome_cleaner_unittests",
+    "//chrome/chrome_cleaner/executables:chrome_cleanup_tool",
+    "//chrome/chrome_cleaner/executables:software_reporter_tool",
+  ]
+
+  if (is_internal_chrome_cleaner_build) {
+    deps += [
+      "${chrome_cleaner_internal_root}:build_targets",
+      "${chrome_cleaner_internal_root}:test_targets",
+    ]
+  }
+}
